src/Computer/GlideComputerAirData.cpp

```diff
@@ -360,39 +361,36 @@ GlideComputerAirData::LastThermalStats(const MoreData &basic,
                                        DerivedInfo &calculated,
                                        bool last_circling)
 {
-  if (calculated.circling || !last_circling ||
-      !calculated.climb_start_time.IsDefined())
-    return;
-
-  auto duration = calculated.cruise_start_time - calculated.climb_start_time;
-  if (duration < THERMAL_TIME_MIN)
+  if (calculated.circling || !last_circling)
     return;
 
-  auto gain = calculated.cruise_start_altitude_te
-    - calculated.climb_start_altitude_te;
-
-  if (gain <= 0)
-    return;
-
-  bool was_defined = calculated.last_thermal.IsDefined();
-
-  calculated.last_thermal.start_time = calculated.climb_start_time;
-  calculated.last_thermal.end_time = calculated.cruise_start_time;
-  calculated.last_thermal.gain = gain;
-  calculated.last_thermal.duration = duration;
-  calculated.last_thermal.start_altitude = calculated.climb_start_altitude_te + (basic.nav_altitude-basic.TE_altitude);
-  calculated.last_thermal.CalculateLiftRate();
-  assert(calculated.last_thermal.lift_rate > 0);
-
-  if (!was_defined)
-    calculated.last_thermal_average_smooth =
-        calculated.last_thermal.lift_rate;
-  else
-    calculated.last_thermal_average_smooth =
-        LowPassFilter(calculated.last_thermal_average_smooth,
-                      calculated.last_thermal.lift_rate, LOW_PASS_FILTER_THERMAL_AVERAGE_ALPHA);
-
-  ThermalSources(basic, calculated, calculated.thermal_locator);
+  // event condition: enter cruise mode, alias leaving climbe mode
+
+  if (calculated.climb_start_time.IsDefined())
+  {
```

Comment:
I don't understand any of these changes. None of this appears to be about clearing TC data.

Reply:
I made the effort to understand the given lines. Here the outcome:
The couple return statements at the beginning of the method do prevent a proper action on all off them applying. That we do need here, it's the  above mentioned assignment of last_thermal, and the Clear() of the current_thermal.
In other words: `(calculated.circling || !last_circling) return;` or better `(!calculated.circling && last_circling) do {}` is just the important transition trigger climb to cruise. That is the routines purpose.
Then we have the condition of gain <=0 or thermal duration to small to trash the current_thermal information and not shift it into last_thermal.
That is what you clearly can read here now. The assignment of the last_thermal was a kludge. current thermal is at this moment well defined and last_thermal should just show exactly that.